src/cdk-experimental/scrolling/auto-size-virtual-scroll.ts

```diff
@@ -355,6 +355,14 @@ export class AutoSizeVirtualScrollStrategy implements VirtualScrollStrategy {
   private _updateTotalContentSize(renderedContentSize: number) {
     const viewport = this._viewport!;
     const renderedRange = viewport.getRenderedRange();
+
+    // once we hit the bottom, we need to resample our range
+    // based on the items in view so we don't create a gap
+    if (renderedRange.end === viewport.getDataLength()) {
```

Comment:
Good to know that this fixes it, probably will want to find a way to do it that doesn't lose our size info though.

Reply:
Ya, I suspected this was more of a starting point rather than the actual fix.